src/components/data-management/SamplesTable.jsx

```diff
@@ -48,6 +50,18 @@ const exampleDatasets = [
   },
 ];
 
+// If a sample exists in projects but not in the samples file,
+// there is an error uploading the sample. We should show an error for that sample.
+const sampleCreateErrorTemplate = {
```

Comment:
do you really need to create `sampleCreateErrorTemplate`? If I look at the code below on lines 243-262, you only really use `sampleCreateErrorTemplate` for:
1. Getting the different name, which will be fixed, because now with saving samples to experiments table (your api PR), you can just display the sample and `samples[sampleUuid]` will be defined.
2.  Displaying the `{ upload: { status: UploadStatus.FILE_NOT_FOUND } },` for the files. But you don't need these fields, because that will be handled in lines 246, 247 of the code.

So you can remove the `sampleCreateErrorTemplate`  and bringing back these changes:
```
      // upload problems sometimes lead to partial updates and incosistent states
      // in this situation it's possible that the sampleUuid does not exist
      // this a temporary fix so that the whole UI doesn't crash preventing the
      // user from removing the dataset or uploading another one.
      // this situation should be included into the SamplesTable.jsx tests
      const sampleFiles = samples[sampleUuid]?.files || {};
```

Reply:
The sample name will only be defined for new samples. Old samples will still throw an error. Would that be ok?

Reply:
I'll implement a simpler default to fill in the data, to cover the case for these old experiments without having to create `sampleCreateErrorTemplate`

Reply:
`Old samples will still throw an error. Would that be ok?`I don't understand the question, can you expand? Can't you just display this error name `'UPLOAD ERROR: Please reupload sample'` for sample name for the old samples that don't have a name?

Reply:
Ah I see your point. There's no need to have `sampleCreateErrorTemplate` because we only need the `name` and `samples`, and the cases where these are empty are already handled when we're generating the data for the table. I agree. I've removed the `sampleCreateErrorTemplate` and assign the default value of the file name to error if the sample record doesn't exist.